src/openms/source/ANALYSIS/ID/IDMapper.cpp

```diff
@@ -229,6 +229,146 @@ namespace OpenMS
+          IntList current_charges;
+          if (!ignore_charge_)
+          {
+            current_charges.push_back(precursors[i_p].getCharge());
+            current_charges.push_back(0); // "not specified" always matches
+          }
+
+          //check if we compare distance from centroid or subelements
+          if (!measure_from_subelements)
+          {
+            if (isMatch_(rt_prec - cmap[cm_index].getRT(), mz_prec, cmap[cm_index].getMZ()) && (ignore_charge_ || ListUtils::contains(current_charges, cmap[cm_index].getCharge())))
+            {
+              was_added = true;
+              if (cmap[cm_index].metaValueExists("spectrum_index"))
+              {
+                IntList indices = cmap[cm_index].getMetaValue("spectrum_index").toIntList();
+                indices.push_back(i);
+                cmap[cm_index].setMetaValue("spectrum_index", indices);
+              }
+              else
+              {
+                IntList il;
+                il.push_back(i);
+                cmap[cm_index].setMetaValue("spectrum_index", il);
+              }
+              ++assigned[i];
+            }
+          }
+          else
+          {
+            for (ConsensusFeature::HandleSetType::const_iterator it_handle = cmap[cm_index].getFeatures().begin();
+                 it_handle != cmap[cm_index].getFeatures().end();
+                 ++it_handle)
+            {
+              if (isMatch_(rt_prec - it_handle->getRT(), mz_prec, it_handle->getMZ())  && (ignore_charge_ || ListUtils::contains(current_charges, it_handle->getCharge())))
+              {
+                was_added = true;
+                if (mapping[cm_index].count(i) == 0)
+                {
+                  if (annotate_with_subelements)
+                  {
+                    // store the map index the precursor was mapped to
+                    Size map_index = it_handle->getMapIndex();
+                    String key = "spectrum_index_" + String(map_index);
+                    if (cmap[cm_index].metaValueExists(key))
+                    {
+                      IntList indices = cmap[cm_index].getMetaValue(key).toIntList();
+                      indices.push_back(i);
+                      cmap[cm_index].setMetaValue(key, indices);
+                    }
+                    else
+                    {
+                      IntList il;
+                      il.push_back(i);
+                      cmap[cm_index].setMetaValue(key, il);
+                    }
+                  }
+                  
+                  ++assigned[i];
+                  mapping[cm_index].insert(i);
+                }
+                break; // we added this precursor already.. no need to check other handles
+              }
+            }
+            // continue to here
+          }
+
+          if (was_added)
+            break;
+
+        } // m/z values to check
+
+        // break to here
+
+      } // features
+    } // Identifications
```

Comment:
```
} // Identifications
```
But right now you only tag precursors to the consensus feature so no identifications? Is that planned?